src/components/list/list.js

```diff
@@ -327,7 +363,9 @@ function mdListItemDirective($mdAria, $mdConstant, $mdUtil, $timeout) {
       function copyAttributes(item, wrapper) {
         var copiedAttrs = $mdUtil.prefixer([
           'ng-if', 'ng-click', 'ng-dblclick', 'aria-label', 'ng-disabled', 'ui-sref',
-          'href', 'ng-href', 'target', 'ng-attr-ui-sref', 'ui-sref-opts'
+          'href', 'ng-href', 'target', 'ng-attr-ui-sref', 'ui-sref-opts',
+          // Custom attributes for the md-button component.
+          'md-no-focus-style'
```

Comment:
@ThomasBurleson This is just forwarding the attribute, because it's not supported by the `md-list-item`, but it should be supported for the button executor.

Actually, I agree with you, it would be \*cooler\* to use a class here, but this will be for now more complicated and will require some additional checks, to only forward this specific class to the button executor.

The attribute itself isn't introduced in this PR, it's just supporting it for the button executor (See #8691)